device/devicegate/filter.go

```diff
@@ -0,0 +1,263 @@
+package devicegate
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"sync"
+
+	"github.com/go-kit/kit/log/level"
+	"github.com/xmidt-org/webpa-common/device"
+	"github.com/xmidt-org/webpa-common/logging"
+	"github.com/xmidt-org/webpa-common/xhttp"
+)
+
+const (
+	metadataMapLocation = "metadata_map"
+	claimsLocation      = "claims"
+)
+
+var (
+	emptyVal = struct{}{}
+)
+
+type FilterGateOption func(*FilterGate)
+
+type FilterGate struct {
+	filters        map[string]map[interface{}]struct{}
```

Comment:
This is complex enough to warrant a custom type.  Also, what purpose is the `map[interface{}]struct{}`?  If you're wanting a set of `interface{}`, a better type declaration is `map[interface{}]bool`.